lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1866,7 +1866,7 @@ private function doMerge($entity, array &$visited, $prevManagedCopy = null, $ass
                 }
             }
 
-            if ($class->isVersioned) {
+            if ($class->isVersioned && !($this->isNotInitializedProxy($managedCopy) || $this->isNotInitializedProxy($entity))) {
```

Comment:
In conjunction with my [other note](https://github.com/doctrine/doctrine2/pull/1573/files#r46593722), I think this conditional would be clearer as:

    $class->isVersioned && $this->isLoaded($managedCopy) && $this->isLoaded($entity)